### DISCUSSION TITLE
Can mfsmount "Prefered Labels" be used for a basic write Tier?

### DISCUSSION THREAD

**Discussion by Blackpaw** [moosefs/moosefs#426] [Q&A]
e.g two small ssd chunkservers be used for preferred writes with rebalance distributing the chunks to a much larger hdd chunkserver pool in the background.

example thought bubble:
Chunkservers:
* CS_SSD1 : 512 GB Label S
* CS_SSD2 : 512 GB Label S
* CS_HDD1 : 5TB LABEL H
* CS_HDD2 : 5TB LABEL H
* CS_HDD2 : 5TB LABEL H

/mnt/mfs : Replication 2 (Class 2)

mfsmount -o mfspreflabels=S /mnt/mfs

The idea is that the initial writes (and reads) will go to the SSD chunkservers first. Being much smaller the chunks will be rebalanced out to the HDD chunkservers in the background.

Does this make sense? or is it horrible hackery that will destroy the system.

Is the 2nd SSD even needed for this to work?

**Comment by jkiebzak** [comment]
First, you have to be careful that your writes do not fill your SSDs faster
than they can replicate the chunks off.

Second, I think you may want to look at mfsscadmin tool
<https://manpages.debian.org/testing/moosefs-client/mfsscadmin.1.en.html>,
which is designed for this scenario. You can specify how much time before
chunks are moved from the "keep" to "archive" label. You could also use
your "s" label for the "create". But note: you cannot specify how long
chunks stay in the "create" mode. As an example you could do:
```mfsscadmin /mnt/mfs create -m LOOSE -C S,S -K S,H -A H,H -d 2 STD```

- `-m LOOSE` - allows chunks to initially be written to other chunk servers
if "S" is overloaded
- `-C S,S` - should initially write two chunks to "S"
- `-K S,H`- will move one chunk to "H" soon after creation
- `-A H,H -d 2 STD` - will move other chunk to "H" 2 days after the initial
chunk creation.

If you use this in combination with your provided "mfsmount" command, I
think you can achieve what you're looking for.

Jason

**Reply by Blackpaw** [reply]
Thanks, thats really interesting and quite powerful. Going to have to read and digest this for a while I suspect. I do love the way moose makes this easy to test on a nominated directory, avoiding messing with the main system.

Any reason as to why you only have one chunk at a time being migrated and the archive option? Wouldn't:

`mfsscadmin /mnt/mfs create -m LOOSE -C S,S -K H,H STD`

Have the effect I was aiming for? (in a safer manner than my original idea)

**Comment by jkiebzak** [comment]
I just provided a "working" example. Many people would opt for more than
"goal 2", in which case you'd want `-C S,S,S`, etc.

Yes, your example also works just as fine. I wasn't sure how long you
wanted chunks to stay on "S". If you don't mind them being quickly moved
off of "S", then your example is good.

**Reply by Blackpaw** [reply]
Thanks, much appreciated. This really expands the potential for moose for me. Wish I'd known this when I was using lizardfs to host VM's
